react/features/transcription/components/TranscriptionSubtitles.web.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link TranscriptionSubtitles}.
+ */
+type Props = {
+
+    /**
+     * Map of transcriptMessageID's with corresponding transcriptMessage.
+     */
+    transcriptMessages: Map<string, Object>
+};
+
+/**
+ * React {@code Component} which can display speech-to-text results from
+ * Jigasi as subtitles.
+ */
+class TranscriptionSubtitles extends Component<Props> {
+
+    /**
+     * Updates the transcription subtitles only if the Map of transcriptMessages
+     * change otherwise prevents the unnecessary re-render..
```

Comment:
Double periods. 

Why did you choose using shouldComponentUpdate over PureComponent? (not saying you should change anything, just asking for your thoughts)

Reply:
I read that PureComponent does shallow comparision and if we are deriving data from the props in the `render()` method, we should avoid PureComponent.

Reply:
It seems to me you're doing a shallow comparison here in shouldComponentUpdate and I don't think you're deriving data in render; you're rendering data but not using existing data to calculate other data. I might be missing what you're saying though so please correct me.

Reply:
Yes, I misunderstood shallow comparisions. In our case, as we are always creating a new map, Pure Component should be fine. 
By deriving data, does it mean that we are using it to pass information to child components?

Reply:
Derived data probably refers to calculating some other state or data or variable using props--deriving a new value from existing values. Passing by itself wouldn't be deriving. I think I found the article you read and the recommendation there was to cache derived data instead of recalculating it in render, especially if what you derive is an object in which case a new object would be created each time causing a re-render.